rest-client/src/main/java/org/thingsboard/rest/client/RestClient.java

```diff
@@ -305,17 +306,28 @@ public void clearAlarm(AlarmId alarmId) {
         restTemplate.postForLocation(baseURL + "/api/alarm/{alarmId}/clear", null, alarmId.getId());
     }
 
-    public PageData<AlarmInfo> getAlarms(EntityId entityId, AlarmSearchStatus searchStatus, AlarmStatus status, TimePageLink pageLink, Boolean fetchOriginator) {
+    public PageData<AlarmInfo> getAlarms(EntityId entityId, AlarmSearchStatus searchStatus, AlarmStatus status, TimePageLink pageLink, UUID offset, Boolean fetchOriginator) {
+        String urlSecondPart = "/api/alarm/{entityType}/{entityId}?fetchOriginator={fetchOriginator}&";
         Map<String, String> params = new HashMap<>();
         params.put("entityType", entityId.getEntityType().name());
         params.put("entityId", entityId.getId().toString());
-        params.put("searchStatus", searchStatus.name());
-        params.put("status", status.name());
+        if(searchStatus != null) {
+            params.put("searchStatus", searchStatus.name());
+            urlSecondPart += "searchStatus={searchStatus}&";
+        }
+        if(status != null) {
+            params.put("status", status.name());
+            urlSecondPart += "status={status}&";
+        }
         params.put("fetchOriginator", String.valueOf(fetchOriginator));
+        if(offset != null) {
+            params.put("offset", offset.toString());
```

Comment:
Please remove this field from the call and remove 
```java
private UUID idOffset;
```
from org.thingsboard.server.common.data.alarm.AlarmQuery since it is not used (and from controller)